src/types/BotContext.ts

```diff
@@ -0,0 +1,18 @@
+import { Client, Collection } from 'discord.js'
+
+import { CommandHandlerConfig } from './CommandHandlerConfig.js'
+
+/**
+ * A shared context object that is passed to all command handlers.
+ */
+export interface BotContext {
+  /**
+   * A Discord.js client instance.
+   */
+  readonly client: Client
```

Comment:
This can be `readonly client: Bot` if we are using `import type Bot`. Just in case we want to have some public method to be accessed by handlers in the future.

Reply:
@baskun that would reintroduce the cyclic dependency 😂

i'm aware that cyclic `import type` will not cause any runtime problem, but when refactoring code, cyclic dependencies make things hard to refactor. it also enables the Bot class to evolve into a "god object". getting rid of cyclic deps helps in preventing this from happening.

in languages like Go, introducing cyclic import cycles becomes a compile error. so now I dread cyclic deps. not only from runtime standpoint but from the software architecture standpoint as well.

> in case we want to have some public method to be accessed by handlers in the future

in this case, we can put that method in the BotContext krub.

Reply:
The one you tagged is not me btw 😂.

Ok kub, I got your point now kub. Then this seems to be the better in also architecture way.
But should we also change the rule in `eslintrc` from 'warn' to 'error'?

Reply:
whoops

it can be either warn or error, ci will fail on both krub 🔥

Reply:
Ok, then it should be good to go krub.